pydicom/dataset.py

```diff
@@ -411,6 +400,8 @@ def __init__(
         self._pixel_array: Optional["numpy.ndarray"] = None
         self._pixel_id: Dict[str, int] = {}
 
+        self.file_meta: FileMetaDataset
```

Comment:
Yes, I think we shall get rid of all these optional attributes (are there any left?) and convert them to proper instance variables.

Reply:
Technically its still an optional attribute. `hasattr(ds, "file_meta")` will return `False`. It's just typed now when it is present.